SECURITY.md

```diff
@@ -0,0 +1,110 @@
+# Security Reporting Process
```

Comment:
Once this PR is merged, we should add a link to these docs to the opentofu website.